internal/controller/authenticator_test.go

```diff
@@ -312,3 +318,92 @@ func (suite *ControllerTestSuite) TestParseToken() {
 	_, err = ParseToken("invalid-token")
 	assert.Error(suite.T(), err)
 }
+
+func (suite *ControllerTestSuite) TestHTTPProxySupport() {
+	// Start a mock proxy server
+	proxyServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write([]byte("Proxy used")); err != nil {
+			suite.T().Errorf("Failed to write response: %v", err)
+		}
+	}))
+	defer proxyServer.Close()
+
+	// Set the HTTPS_PROXY environment variable to the mock proxy
+	os.Setenv("HTTPS_PROXY", proxyServer.URL)
+	defer os.Unsetenv("HTTPS_PROXY")
+
+	// Use a mock target server
+	targetServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write([]byte("Target server reached")); err != nil {
+			suite.T().Errorf("Failed to write response: %v", err)
+		}
+	}))
+	defer targetServer.Close()
+
+	tlsConfig := &tls.Config{InsecureSkipVerify: true}
+	proxyURL, _ := url.Parse(proxyServer.URL)
+	transport := &http.Transport{
+		TLSClientConfig: tlsConfig,
+		Proxy:           http.ProxyURL(proxyURL), // Explicitly set the proxy
+	}
+	client := &http.Client{Transport: transport}
+
+	// Log the proxy URL and environment variable for debugging
+	fmt.Println("Proxy URL:", proxyServer.URL)
+	fmt.Println("HTTPS_PROXY Env:", os.Getenv("HTTPS_PROXY"))
```

Comment:
```suggestion
	// Log the proxy URL and environment variable for debugging
	suite.T().Logf("Proxy URL: %s", proxyServer.URL)
	suite.T().Logf("HTTPS_PROXY Env: %s", os.Getenv("HTTPS_PROXY"))
```
Same for all the other usages of `fmt.Print*` in tests in this PR.